packages/gatsby-cli/src/create-cli.js

```diff
@@ -315,7 +316,17 @@ module.exports = argv => {
     .wrap(cli.terminalWidth())
     .demandCommand(1, `Pass --help to see all available commands and options.`)
     .strict()
-    .showHelpOnFail(true)
-    .recommendCommands()
+    .fail((msg, err, yargs) => {
+      const availableCommands = yargs.getCommands().map(commandDescription => {
+        const [command] = commandDescription
+        return command.split(` `)[0]
+      })
+      const arg = argv.slice(2)[0]
+      const suggestion = arg ? didYouMean(arg, availableCommands) : ``
+
+      cli.showHelp()
+      report.log(suggestion)
+      report.log(msg)
```

Comment:
Should be log the `msg` as well? 

```
~/Code/gatsby/www feat/gatsby-cli-suggestions*
❯ ./node_modules/gatsby-cli/lib/index.js ne .
Usage: node_modules/gatsby-cli/lib/index.js <command> [options]

Commands:
  develop                   Start development server. Watches files, rebuilds, and hot reloads if something changes
  build                     Build a Gatsby project.
  serve                     Serve previously built Gatsby site.
  info                      Get environment information for debugging and issue reporting
  clean                     Wipe the local gatsby environment including built assets and cache
  repl                      Get a node repl with context of Gatsby environment, see (add docs link here)
  new [rootPath] [starter]  Create new Gatsby project.

Options:
  --verbose      Turn on verbose output                                                                                                                                       [boolean] [default: false]
  --no-color     Turn off the color in output                                                                                                                                 [boolean] [default: false]
  -h, --help     Show help                                                                                                                                                                     [boolean]
  -v, --version  Show version number                                                                                                                                                           [boolean]


Did you mean this?
     new

Unknown arguments: ne, .
```

In this case, maybe we can do without the `Unknown arguments` bit. What do you think?

Reply:
Seems like this is good to have for future proofing as correctly pointed out by @pieh

Reply:
Just to let you know, I decided to log the `msg` as `Unknown arguments: <args>` as it clearly states the mistakes the user has made.

This will help Newcomers and Junior Developers like myself take up Gatsby faster. 

The ethos here is copy a **Rust Lang** level of stderr, as to makes it so obvious, that it makes **Gatsby** easier to adopt

Nonetheless I've been thinking of some improvements, that can be made to make the `Unknown arguments: ` part more readable and aesthically pleasing to Gatsby Commnity

I've already got some ideas, plus it would be nice to hear back on what you guys think 😄